src/mudlet.cpp

```diff
@@ -2481,32 +2486,39 @@ void mudlet::playSound(QString s, int soundVolume)
     if (!pPlayer) {
         pPlayer = new QMediaPlayer(this);
 
-        auto pHost = getActiveHost();
 
         if (!pPlayer) {
             /* It (should) be impossible to ever reach this */
-            if (pHost) {
-                pHost->postMessage("\n[  ERROR  ]  - Unable to create new QMediaPlayer object\n");
-            }
+            pHost->postMessage("\n[  ERROR  ]  - Unable to create new QMediaPlayer object\n");
             return;
         }
 
-        connect(pPlayer, &QMediaPlayer::stateChanged, [=](QMediaPlayer::State state) {
-            if (state == QMediaPlayer::StoppedState) {
-                TEvent soundFinished;
-                soundFinished.mArgumentList.append("sysSoundFinished");
-                soundFinished.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
-                soundFinished.mArgumentList.append(pPlayer->media().canonicalUrl().fileName());
-                soundFinished.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
-                soundFinished.mArgumentList.append(pPlayer->media().canonicalUrl().path());
-                soundFinished.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
-                pHost->raiseEvent(soundFinished);
-            }
-        });
-
         mMusicBoxList.append(pPlayer);
     }
 
+    // Remove any previous connection to the signal of this QMediaPlayer,
+    // theoretically this might be movable to be within the lambda function of
+    // the following connect(...) but that does seem a bit twisty and this works
+    // well enough!
+    disconnect(pPlayer, &QMediaPlayer::stateChanged, 0, 0);
```

Comment:
Aren't you glad Qt API isn't spewing out warnings if there wasn't any connection to begin with? :wink: